### DISCUSSION TITLE
What happens when you sell or get rid of a car?

### DISCUSSION THREAD

**Discussion by sergiozygmunt** [teslamate-org/teslamate#3283] [Q&A]
Sorry if already asked - I want to confirm current behavior if a vehicle is removed from my Tesla account. Does the data get dropped or can I continue to access historical data forever?

**Comment by ktsenvolt** [comment]
Once you delete  Tesla account and you will no longer able to login to your Teslamate.

**Reply by sergiozygmunt** [reply]
What happens to historical data? Isn't it kept in the local database?

**Comment by ktsenvolt** [comment]
I don't know how to access to local database if Teslamate could not been login.

**Reply by sergiozygmunt** [reply]
There is no login page for Teslamate itself --- it just runs and your reverse proxy puts a login page. Grafana has its own login which is different from Teslamate and the Tesla API.

**Comment by sergiozygmunt** [comment]
I figured out what happened on my own. Histotical data (i.e. the data you access in Grafana) is kept even if the car is removed from your Tesla account.
